dali/driver/hasseb_firmware/README.md

```diff
@@ -0,0 +1,219 @@
+  1 for response.
+  
+* Byte 5: Reserved
+
+* Byte 6: If set to other value than 0, the message will be sent twice. The
+  value in byte 6 determines the delay between messages in milli seconds.
+
+* Bytes 7-9: Data bytes sent to DALI bus.
+
+Reports from interface to host
+------------------------------
+
+* Byte 0: Preamble, always 0xAA
+* Byte 1: Report type
+* Byte 2: Sequence number
+* Bytes 3–9: Report dependent
+
+Reports generated in response to a command from the host will include
+the sequence number of the command in byte 2. Reports generated
+otherwise (for example in response to bus activity) byte 2 is set to
+zero. Commands with sequence number zero are not allowed from host.
+
+### List of report types
+
+* 0x00: No operation, ignore this report
+* 0x01: Hardware type
+* 0x02: Firmware version
+* 0x03: Device serial number
+* 0x04: Bus status
+* 0x05: Unused
+* 0x06: Bus configuration
+* 0x07: Transmission report
+
+### Report descriptions
+
+#### Hardware type, NOT IMPLEMENTED
+
+* Byte 3: capability bits
+  * Bit0: Has internal power supply
+  * Bit1: Internal power supply can be switched in software
+  * Bit2: Can detect bus over voltage
+* Bytes 4–n: reserved, zero
+
+#### Firmware version
+
+* Byte 3: major version
+* Byte 4: minor version
+
+#### Device serial number, NOT IMPLEMENTED
+
+* Byte 3: length of serial number in bytes; zero means "unsupported"
+* Byte 4–n: serial number, LSB first
+
+#### Bus status, NOT IMPLEMENTED
+
+This report can be sent spontaneously when the bus status changes, as
+well as in response to a "read bus status" command.
+
+* Byte 3: detected idle bus voltage in tenths of a volt, or zero if
+  not implemented
+
+* Byte 4: status bits
+  * Bus shorted
+  * Bus over voltage
+
+#### Bus configuration, NOT IMPLEMENTED
+
+* Byte 3: internal power supply maximum current. If configurable,
+  this is the actual maximum current. If no internal power supply is
+  present, this is zero. If there is an internal power supply but its
+  current is set in hardware and not known in firmware, this is 255.
+
+* Byte 4: internal power supply voltage. If configurable, this is the
+  actual voltage in tenths of a volt. If there is an internal power
+  supply but its voltage is set in hardware and not known in firmware,
+  this is 255. If no internal power supply is present, this is zero.
+
+* Byte 5: flags
+  * Bit 0: Bus short circuited.
+
+#### Transmission report
+
+A report sent to host when a DALI message is received from the bus.
+
+* Byte 2: Sequence number. If the report is a backward frame, the sequence
+  number equals the sequence number of the forward frame. For a 
+  sniffing byte, the sequence number is zero.
+  
+* Byte 3: Report type
+  * 0x01: No response received
+  * 0x02: Response data received
+  * 0x03: Invalid data
+  * 0x04: Answer too early
+  * 0x05: Sniffing byte
+  * 0x06: Sniffing byte error
+
+* Byte 4: Data length in bits
+
+* Byte 5-7: Data
+
```

Comment:
I'm a bit confused about this protocol.  I was expecting something like:
* host -> device: send *this*, I'm expecting a response
* device -> host: I sent it!
* device -> host: I got a response!

And for send-twice commands:
* host -> device: send *this* twice
* device -> host: I sent it!
* device -> host: I sent it again!

It looks like there's no reply to the host at all for commands that don't expect a response?

Reply:
By the way, is there a github project for the firmware where I can submit issues?  I need to say things like "the string descriptors are faulty" but it's not really on-topic here.

Reply:
@sde1000 @hasseb I already stated that IMHO firmware related binaries and documentation belongs to hasseb org , best inside an own project, not here. thanks

Reply:
I definitely don't want to include binaries, but I think there's a case for including documentation about how to drive the various interfaces.  I might want to include my notes on driving the Tridonic/Lunatone DALI USB interface, for example.

Reply:
sure, but i think such docs belongs to the doc strings in the related driver implementation.